src/main/java/com/fasterxml/jackson/core/StreamReadConfig.java

```diff
@@ -0,0 +1,28 @@
+package com.fasterxml.jackson.core;
+
+/**
+ * The configuration for streaming reads.
+ *
+ * @since 2.15
+ */
+public class StreamReadConfig {
+
+    private static final int DEFAULT_MAX_NUM_LEN = 1000;
+
+    private int _maxNumLen = DEFAULT_MAX_NUM_LEN;
```

Comment:
Need to also consider the fact that as-is this would be specific to textual formats. Binary formats would either:

1. Interpret this as byte limit instead of char (which is probably fine), or
2. use different setting

Either way would need to be documented once we figure out way to go.

Reply:
currently all the existing code takes Strings and parses Strings - everything has been converted to text

Reply:
I understand this is the case for default handling, and in 2.x. Just noting that these checks would not be applied by, say, CBOR or Smile parsers as they use binary encoding of numbers and do not convert to `String`s in most cases.

But I think it'd make sense to consider length in "units of input", which can then be chars or bytes as appropriate.